testsuite/spicedb/spicedb.py

```diff
@@ -85,53 +87,195 @@ def create_schema(self, schema_config: SchemaConfig):
+            "namespace": self.cluster.project,
+        }
+
+    @property
+    def grpc_port(self):
+        """Returns the gRPC port number"""
+        return self.service.get_port("grpc").port
+
+    @property
+    def http_port(self):
+        """Returns the HTTP port number"""
+        return self.service.get_port("http").port
+
+    def set_gateway_url(self, gateway_url: str):
+        """
+        Set the Gateway URL for accessing SpiceDB externally.
+        """
+        self._gateway_url = gateway_url
+
+    @property
+    def http_url(self):
+        """
+        Returns the HTTP API URL for SpiceDB client (test suite access).
+        """
+        return self._gateway_url
+
+    @property
+    def grpc_url(self):
+        """
+        Returns the internal cluster DNS with gRPC port for pod-to-pod communication.
+        """
+        return f"{self.name}.{self.cluster.project}.svc.cluster.local:{self.grpc_port}"
+
+    @property
+    def client(self):
+        """
+        Lazy-initialized SpiceDB client for managing schemas and relationships.
+        Uses HTTP API for external access (test suite via Gateway).
+        """
+        if self._client is None:
+            self._client = SpiceDBClient(self.http_url, self.preshared_key)
+        return self._client
+
+    def commit(self):
+        """Deploy SpiceDB to Kubernetes"""
+        match_labels = {"app": self.label, "deployment": self.name}
+
+        self.deployment = Deployment.create_instance(
+            self.cluster,
+            self.name,
+            container_name="spicedb",
+            image=self.image,
+            ports={"grpc": 50051, "http": 8443},
+            selector=Selector(matchLabels=match_labels),
+            labels={"app": self.label},
+            command_args=[
+                "serve",
+                "--grpc-preshared-key",
+                self.preshared_key,
+            ],
         )
-        response = self.client.CheckPermission(check_request)
-        return response.permissionship == response.PERMISSIONSHIP_HAS_PERMISSION
+        env_vars = self.deployment.container.setdefault("env", [])
+        env_vars.append({"name": "SPICEDB_HTTP_ENABLED", "value": "true"})
+        env_vars.append({"name": "SPICEDB_LOG_LEVEL", "value": "debug"})
+        self.deployment.commit()
+        self.deployment.wait_for_ready()
+
+        self.service = Service.create_instance(
+            self.cluster,
+            self.name,
+            selector=match_labels,
+            ports=[
+                ServicePort(name="grpc", port=50051, targetPort="grpc"),
+                ServicePort(name="http", port=8443, targetPort="http"),
+            ],
+            labels={"app": self.label},
+        )
+        self.service.commit()
+
+    def delete(self):
+        """Delete SpiceDB deployment and service from Kubernetes"""
+        if self.deployment:
+            self.deployment.delete()
+        if self.service:
+            self.service.delete()
```

Comment:
There is a delete method you inherit from Backend class that I think has more complete cleanup process for the backend services